include/circt/Dialect/LTL/LTLTypes.td

```diff
@@ -0,0 +1,50 @@
+//===- LTLTypes.td - LTL dialect types ---------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_LTL_LTLTYPES_TD
+#define CIRCT_DIALECT_LTL_LTLTYPES_TD
+
+include "circt/Dialect/LTL/LTLDialect.td"
+include "mlir/IR/AttrTypeBase.td"
+
+class LTLTypeDef<string name, string typeMnemonic> : TypeDef<LTLDialect, name> {
+  let mnemonic = typeMnemonic;
+}
+
+def LTLSequenceType : LTLTypeDef<"Sequence", "sequence"> {
+  let summary = "LTL sequence type";
+  let description = [{
+    The `ltl.sequence` type is returned by any of the `ltl.seq.*` operations. It
+    represents a snippet of linear temporal logic, for example, *"A is true two
+    cycles after B is true"*.
```

Comment:
I don't like the informal phrasing "snippets" here.

Reply:
Good point, let me tweak this a bit.